src/box/lua/connpool.lua

```diff
@@ -196,19 +218,33 @@ local function filter(opts)
         instances = '?table',
         labels = '?table',
         roles = '?table',
+        sharding_roles = '?table',
         mode = '?string',
     })
     opts = opts or {}
     if opts.mode ~= nil and opts.mode ~= 'ro' and opts.mode ~= 'rw' then
         local msg = 'Expected nil, "ro" or "rw", got "%s"'
         error(msg:format(opts.mode), 0)
     end
+    if opts.sharding_roles ~= nil then
+        for _, sharding_role in ipairs(opts.sharding_roles) do
+            if sharding_role ~= 'storage' and
+               sharding_role ~= 'rebalancer' and
+               sharding_role ~= 'router' then
+               local msg = 'Unknown sharding role \"%s\" in '..
+                           'connpool.filter() call. Expected one of the '..
+                           '\"storage\", \"router\", \"rebalancer\"'
+               error(msg:format(sharding_role), 0)
+            end
+        end
+    end
```

Comment:
I would exclude `rebalancer`, because it is not very trivial and I guess that it is not really needed by users.

The default `sharding.rebalancer_mode` is `'auto'`, which means that the rebalancer chooses an appropriate instance on its own (some of the RW storage instances). It means that the option is not always static and, moreover, usually dynamic. We can track where the rebalancer dynamically (like we do for RO/RW mode), but I think that it doesn't worth the effort.

At the same time, supporting of the static rebalancer configuration without supporting the dynamic one would look confusing, as for me. I would just raise an error that `sharding_role = "rebalancer"` is not supported (this error should be different from the unknown sharding role error).